packages/netlify-cms-backend-azure/src/API.ts

```diff
@@ -0,0 +1,1007 @@
+          scopePath: path,
+          recursionLevel: 'full',
+        };
+      } else {
+        azureGitItemParams = {
+          version: this.branch,
+          path,
+          recursionLevel: 'none',
+        };
+      }
+
+      const azureGitItem = await this.requestJSON<AzureGitItem>({
+        url: `${this.endpointUrl}/items/`,
+        params: azureGitItemParams, // Azure
+      });
+
+      const azureGitTreeRef = await this.requestJSON<AzureGitTreeRef>(
+        azureGitItem._links.tree.href,
+      );
+
+      const azureTreeEntries = azureGitTreeRef.treeEntries || [];
+      if (!Array.isArray(azureTreeEntries)) {
+        throw new Error(
+          `Cannot list files, path ${path} is not a directory but a ${typeof azureTreeEntries}`,
+        );
+      }
+
+      const processedAzureTreeEntries: AzureGitTreeEntryRef[] = [];
+
+      for (const f of azureTreeEntries) {
+        f.relativePath = `${path}/${f.relativePath}`;
+        let entry: AzureGitTreeEntryRef | undefined = f;
+        // If AzureGitTreeEntryRef is still a tree object, we need to drill further to get to the blob
+        if (f.gitObjectType === 'tree') {
+          const azureGitNestedTreeRef = await this.requestJSON<AzureGitTreeRef>(f.url);
+
+          entry = azureGitNestedTreeRef.treeEntries
+            ? azureGitNestedTreeRef.treeEntries[0]
+            : undefined;
+
+          if (entry) {
+            entry.relativePath = `${f.relativePath}/${entry.relativePath}`;
+          }
+        }
+        if (entry) {
+          processedAzureTreeEntries.push(entry);
+        }
+      }
+
+      return processedAzureTreeEntries;
+    } catch (error) {
+      console.error(error);
+      return [];
+    }
+  };
+
+  /**
+   * Gets an AzureRef representing the HEAD commit of the specified branch.
+   * @param branch The name of the branch to get a commit ref for.
+   */
+  async getRef(branch: string = this.branch): Promise<AzureRef> {
+    const refs: { value: AzureRef[] } = await this.requestJSON({
+      url: `${this.endpointUrl}/refs`,
+      params: {
+        $top: '1', // There's only one, so keep the payload small
+        filter: 'heads/' + branch,
+      },
+    });
+
+    return first(
+      refs.value.filter((b: AzureRef) => b.name == this.branchToRef(branch)),
+    ) as AzureRef;
+  }
+
+  async deleteRef(ref: AzureRef): Promise<void> {
+    const deleteBranchPayload = [
+      {
+        name: ref.name,
+        oldObjectId: ref.objectId,
+        newObjectId: '0000000000000000000000000000000000000000',
+      },
+    ];
+
+    await this.requestJSON({
+      method: 'POST',
+      url: `${this.endpointUrl}/refs`,
+      body: JSON.stringify(deleteBranchPayload),
+    });
+  }
+
+  uploadAndCommit(
+    items: AzureCommitItem[],
+    comment = 'Creating new files',
+    branch: string = this.branch,
+    newBranch = false,
+  ) {
+    return this.getRef(branch).then(async (ref: AzureRef) => {
+      if (ref == null) {
```

Comment:
Shouldn't this be an error? If `branch` was passes we want to make sure to commit to it and not to the default one.
If `branch` was not passes it defaults to the default one which in that case calling `getRef` again with the same value is redundant.